spyder/utils/qthelpers.py

```diff
@@ -76,6 +81,63 @@ def event(self, event):
         return QApplication.event(self, event)
 
 
+def get_origin_filename():
+    """Return the filename at the top of the stack"""
+    # Get top frame
+    f = sys._getframe()
+    while f.f_back is not None:
+        f = f.f_back
+    return f.f_code.co_filename
+
+
+if sys.platform == "darwin":

```

Comment:
I think there's no need to define these functions only for macOS. Instead you can group all functions and classes that apply only to macOS and add a block comment above them like this:

```python
# =============================================================================
# Only for macOS
# =============================================================================
```